File: text-embeddings/text.js
import { TwelveLabs } from 'twelvelabs-js';

(async () => {
  const client = new TwelveLabs({ apiKey: process.env.API_KEY}); 
  const engineName = 'Marengo-retrieval-2.6';
  const embedding = await client.embed.create({
    engineName: engineName,
    text: 'Man wearing a hat',
    textTruncate: 'start',
  });
  console.log(`  Engine: ${embedding.engineName}`);
  console.log(`  Embedding: ${embedding.textEmbedding.float}`) 
})();

File: text-embeddings/text.py
import os
from twelvelabs import TwelveLabs

client = TwelveLabs(api_key=os.environ.get("API_KEY"))

embedding = client.embed.create(
  engine_name="Marengo-retrieval-2.6",
  text="Man weearing a hat",
  text_truncate="start",
)

print(f" Engine: {embedding.engine_name}")
print(f" Embedding: {embedding.text_embedding.float}")
